doc/neps/nep-0029-deprecation_policy.rst

```diff
@@ -0,0 +1,294 @@
+   ``python_requires`` in ``setup``.  All supported versions of
+   Python will be in the test matrix and have binary artifacts built
+   for releases.
+
+   The project will bump (adjust upward) the minimum Python and NumPy
+   version support on every minor and major release, but never on a
+   patch release.
+
+For other dependencies, adopt similar time windows of 24 months or
+shorter.
+
+
+Backward compatibility
+----------------------
+
+No backward compatibility issues.
+
+Alternatives
+------------
+
+Ad-Hoc version support
+~~~~~~~~~~~~~~~~~~~~~~
+
+A project could on every release evaluate whether to increase
+the minimum version of Python supported.
+As a major downside, an ad-hoc approach makes it hard for downstream users to predict what
+the future minimum versions will be.  As there is no objective threshold
+to when the minimum version should be dropped, it is easy for these
+version support discussions to devolve into [bike shedding](https://en.wikipedia.org/wiki/Wikipedia:Avoid_Parkinson%27s_bicycle-shed_effect) and acrimony.
+
+
+All CPython supported versions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The CPython supported versions of Python are listed in the Python
+Developers Guide and the Python PEPs. Supporting these are a very
+clear and conservative approach.  However, it means that there is 4
+year lag between when new language features come into the language and
+when the projects are able to use them.  Additionally, for projects
+that have a significant component of compiled extensions this requires
+building many binary artifacts for each release.
+
+For the case of NumPy, many projects carry workarounds to bugs that
+are fixed in subsequent versions of NumPy.  Being proactive about
+increasing the minimum version of NumPy will allow downstream
+packages to carry fewer version-specific patches.
+
+
+
+Default version on Linux distribution
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The policy could be to support the version of Python that ships by
+default in the latest Ubuntu LTS or CentOS/RHEL release.  However, we
+would still have to standardize across the community which
+distribution we are following.
+
+By following the versions supported by major Linux distributions, we
+are giving up technical control of our projects to external
+organizations that may have different motivations and concerns than we
+do.
+
+N minor versions of Python
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Given the current release cadence of the Python, the proposed time
+(42 months) is roughly equivalent to "the last two" Python minor
+versions.  However, if Python changes their release cadence, any rule
+based on the number of minor releases will need to be changed.
+
+
+Time window on the X.Y.1 Python release
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As the first bug fix release is typically a few months after the
+initial release, you can achieve the same effect by using a large delay
+from the X.Y.0 release which seems simpler to explain.
```

Comment:
I still have a hard time understanding this one. I'm assuming "a large delay from the X.Y.0 release" refers to the 6 month buffer on top of the 2 x 18 month time-based rule? 

Let's at least keep the language consistent with the definitions above. Consider: 
"As the first **patch version** is typically released within a few months of the corresponding **minor version**,  a rule based on the first patch release is effectively similar to the proposed rule, given the proposed rule's six-month buffer."

(This isn't strictly true if the release cadence were to change, right?)

One option is to remove this alternative entirely, unless you think it would be missed. If so, do you need to mention rules based on X.Y.2? X.Y.3? (Probably not.)

Reply:
Tried re-writing this again, if it is still not clear we can axe it!